### DISCUSSION TITLE
How to unprotect master?

### DISCUSSION THREAD

**Discussion by BlackWoxs** [python-gitlab/python-gitlab#1991] [Q&A] [answered]
In the previous version of Python-GitLab after creating the project and adding a user as a developer, we have been using:

```
# Unprotect master
branch = project.branches.get('master')
branch.unprotect()
as we wish to allow developer access to the master.
```

unprotect is no longer available in version 3.2.0 can anyone guide me on how to achieve the same result using the new version

Thank you

**Comment by nejch** [comment, marked as answer]
I believe I just answered this on stackoverflow but copy/pasting for posterity. (https://stackoverflow.com/questions/72038367/python-gitlab-unprotect-master)

---

You need to use the [Protected Branches API](https://docs.gitlab.com/ee/api/protected_branches.html#protected-branches-api) ([python-gitlab docs](https://python-gitlab.readthedocs.io/en/stable/gl_objects/protected_branches.html)) for this.


```
protected_branch = project.protectedbranches.get("master")
protected_branch.delete()

# or simply
project.protectedbranches.delete("master")
```

Note: the `delete()` method in this case simply unprotects it, it does not delete the branch.

To re-protect it, you can then create it again. You can also control access levels directly:

```
protected_branch = project.protectedbranches.create({
    'name': 'master',
    'merge_access_level': gitlab.const.DEVELOPER_ACCESS,
    'push_access_level': gitlab.const.MAINTAINER_ACCESS
})

**Reply by BlackWoxs** [reply]
Thank you so much for the quick response and solution - VERY much appreciated.
